lib/services/analytics/analytic_service.dart

```diff
@@ -0,0 +1,230 @@
+        name: "Manage Address",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future order({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "Order",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future changeUsername({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "Change Usename",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future settings({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "Settings",
+        parameters: <String, dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+
+  Future privacyPolicy({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "Privacy Policy",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future notification({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "ToggleNotification",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future termsAndConditions({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "TermsConditions",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future getUserDetails({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "GetUserDetails",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future faq({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "FAQ",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future successPayment({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "SuccessfulPayment",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future gitWebView({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "GitWebView",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future drawer({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "MenuOpened",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future visitCartClicked({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "VisitCart",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future cartScreen({@required String userId}) async {
+    await _analytics.logEvent(
+        name: "CartScreen",
+        parameters: <String,dynamic>{
+          'userId': userId,
+        }
+    );
+  }
+  Future setUserProperties({@required String userId}) async {
+    await _analytics.setUserId(userId);
+  }
+
+}
```

Comment:
No need to pass the userId for every function. Get the userId once and send it to analytics.